newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalProjectWriter.js

```diff
@@ -142,9 +168,20 @@ export const onSaveProject = (
   };
 
   const projectPath = path.dirname(filePath);
-  return writeProjectFiles(project, filePath, projectPath).then(() => {
-    return { wasSaved: true, fileMetadata: newFileMetadata }; // Save was properly done
-  });
+
+  try {
+    deleteExistingFilesFromDirs(project, projectPath);
+  } catch (e) {
+    return Promise.reject(
```

Comment:
I think that deleting the existing files from the folders should fail silently.
You can `console.warn` a message but stopping a save could mean preventing a user from saving if an error happens.

Reply:
Also, since you made the change to an async function (thank you!), could you replace the `return new Promise.reject` line 158 by a `throw new Error('...')`

Reply:
Done !